src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -52,6 +54,50 @@ private FluidUtil()
     }
 
     /**
+     * Used to handle the common case of a player holding a fluid item and right-clicking on a fluid handler block.
+     * First it tries to fill the item from the block,
+     * if that action fails then it tries to drain the item into the block.
+     * Automatically updates the item in the player's hand and stashes any extra items created.
+     *
+     * @param player The player doing the interaction between the item and fluid handler block.
+     * @param hand   The player's hand that is holding an item that should interact with the fluid handler block.
+     * @param world  The world that contains the fluid handler block.
+     * @param pos    The position of the fluid handler block in the world.
+     * @param side   The side of the block to interact with. May be null.
+     * @return true if the interaction succeeded and updated the item held by the player, false otherwise.
+     */
+    public static boolean interactWithFluidHandler(@Nonnull EntityPlayer player, @Nonnull EnumHand hand, @Nonnull World world, @Nonnull BlockPos pos, @Nullable EnumFacing side)
+    {
+        Preconditions.checkNotNull(player);
+        Preconditions.checkNotNull(hand);
+        Preconditions.checkNotNull(world);
+        Preconditions.checkNotNull(pos);
+
+        ItemStack heldItem = player.getHeldItem(hand);
+        if (!heldItem.isEmpty())
+        {
+            IFluidHandler blockFluidHandler = getFluidHandler(world, pos, side);
+            if (blockFluidHandler != null)
+            {
+                IItemHandler playerInventory = new InvWrapper(player.inventory);
```

Comment:
I'm not sure that's equivalent to what I want. 
In fact this should probably be `new InvWrapper(player.inventory.mainInventory)`
I'm going to test it a bit.